JavaScript Task-Group By Assignment

arbabkhanocloud opened the pull request:
This is JavaScript Task#1. Which is converting an array of objects into a single object having keys based on Cities. I have implemented this task with for loop, map(), and reduce functions.

arbabkhanocloud committed:
JS Group By Assignment

This is JavaScript Task#1. In which converting an array of objects into a single object having a key based on Cities,

arbabkhanocloud committed:
Changed code identation

Removed extra lines plus did code indentation.

arbabkhanocloud committed:
outPut of GroupBY Assignment

This is the output of the GroupBy Task

Shazil2154 commented:
@arbabkhanocloud can you add another variation without using the reduce function just add and other file here

arbabkhanocloud committed:
Assignment Variation With Map and ForEach functions

I have added two more variation of GroupByAssignment with Map function and forEach function.

arbabkhanocloud committed:
GroupByAssignment with For Loop

I have added a new variation of GroupByAssignment with for Loop

arbabkhanocloud commented:
> @arbabkhanocloud can you add another variation without using the reduce function just add and other file here

Yes, I have added 3 variations with map(), forEach(), and with simple for loop function. kindly check them out.

mahmad35 commented on GroupByAssignment_forEach.js:
```diff
@@ -0,0 +1,22 @@
+      const emptyObject ={};
+      array.forEach(element => {
+        if((element[key] in emptyObject)===false) {  /*check if a key is already exist or does not exist in object_acumulator */
+              emptyObject[element[key]] = emptyObject[element[key]]   /* making key of object */ 
```
You can write it as 
const elementKey  = element[key]
if (!(elementKey in emptyObject)) {
 emptyObject[elementKey] = []
}

mahmad35 commented on GroupByAssignment_forLoop.js:
```diff
@@ -0,0 +1,22 @@
+    function func(array,key){
+        const emptyObject ={};
+        for( let element =0;  element < array.length; element++){
+            if((array[element][key] in emptyObject)===false) {  /*check if a key is already exist or does not exist in object_acumulator */
```
Similar to last task:

const elementKey = array[ind][key]
if (!(elementKey in emptyObject)) {
emptyObject[elementKey] = []
}

It is easier to read emptyObject[elementKey] than emptyObject[array[element][key]]

mahmad35 commented on GroupByAssignment_reduce.js:
```diff
@@ -0,0 +1,24 @@
+  function func(array,key){
+      return array.reduce((emptyObject, element)=>{                
+          if((element[key] in emptyObject)===false) {  /*check if a key is already exist or does not exist in object_acumulator */
+            emptyObject[element[key]] = emptyObject[element[key]]   /* making key of object */ 
```
Same as last task!

mahmad35 commented on GroupByAssignment_with_Map.js:
```diff
@@ -0,0 +1,23 @@
+
+    function func(array,key){
+      const emptyObject ={};
+        array.map((element)=>{
```
This is not a valid example for using maps. Maps are generally used when we want to map the element of an array to some other structure or format and still return all the data.

arbabkhanocloud replied:
I know array.reduce( ) is the best option here. but I implemented it with map() just for practice purposes.

arbabkhanocloud committed:
Update GroupByAssignment_forLoop.js

removed extra step of making object key. And directly assigning empty array to object key's value.

arbabkhanocloud committed:
Update GroupByAssignment_reduce.js

Removed extra step of making object key. And directly assigning empty array to object key's value.

arbabkhanocloud committed:
Update GroupByAssignment_with_Map.js

Removed extra step of making object key. And directly assigning empty array to object key's value.

arbabkhanocloud committed:
Delete GroupByAssignment_forEach.js

arbabkhanocloud committed:
Removed map() file

I removed map() implementation as it is not suitable.
And I added forEach() file
